### DISCUSSION TITLE
v2.4.1 breaks multi-instance HA discovery

### DISCUSSION THREAD

**Discussion by embcla** [zwave-js/zwave-js-ui#947] [Q&A]
Hi

Something broke in my HA setup that was working for ages, and it got me started on a hunt.
Which led me to this change https://github.com/zwave-js/zwavejs2mqtt/pull/920/files

Basically the HA discovery topic is now made of **only** the nodeID.
Which is unfortunate as when multiple instances push to the same broker, then they most probably have nodes with the same nodeID.
This ends up creating frankenstein objects in HA that contain a mix of features from each device with the same nodeID from each connected instance. But if the two devices have a topic/value in common, then HA can't discern which is which.

Can this be rolled back/fixed/amended somehow?
The previous solution with the name was much better because it was using also the instance name in the topic, thereby making it unique.
Also this makes my multi-instance setup basically non working unless I remove and re-add nodes on one of the instances to have unique nodeIDs

**Comment by robertsLando** [comment]
Rolling it back now, wait for 3.0.2
